Feature/shoot controls

Jazy5552 opened the pull request:
Implemented shooting controls by use of the InputController and the ShootController.

Jazy5552 committed:
Add gun to main player prefab

Also add slippery physics material so the player
doesnt get stuck walking along walls

Jazy5552 committed:
Remove entity folder

This folder simply had a duplicate of the player prefab
which is inside the Player folder.

Jazy5552 committed:
Change to linear coloring

Instead of gamma, this increases the lighting quality of the
game, at the cost of not running on older consoles.

Jazy5552 committed:
Add new Fire input entries

Jazy5552 committed:
Implement simple shoot logic

Jazy5552 committed:
Implement firing input handler logic

Jazy5552 committed:
Add future todo note

Jazy5552 committed:
Implement shooting logic in the shootcontroller

Also remove dependancy on the Player component

Jazy5552 committed:
Add future todo notes

Jazy5552 committed:
Remove hardcoded projectile tag

Further allowing shoot controller be reused by other entities

Jazy5552 committed:
Update the main player to use the new shoot controller

Jazy5552 committed:
Update my  sandbox

Jazy5552 committed:
Add momentum to BasicProjectile

This is added mostly as a test to see if we like the projectile
momentum.

Jazy5552 committed:
Change file to LF line endings

Jazy5552 committed:
Update my sandbox

AleksandrHovhannisyan reviewed (changes requested):
Not sure I like the momentum, it's kinda weird if you move backwards and shoot because then the projectiles are actually slower. What do you think?

Other than that, looks good :) Be sure to resolve the conflict.

AleksandrHovhannisyan commented on Assets/Scripts/Entity/Player/InputController.cs:
```diff
@@ -2,19 +2,31 @@
+[RequireComponent(typeof(ShootController))]
 public class InputController : MonoBehaviour
 {
+    private Player p;
```
I would use more descriptive names. I know it's faster to develop if you use shorter variable names, but `p` isn't very informative :D

Jazy5552 replied:
Fair enough.

AleksandrHovhannisyan commented on Assets/Scripts/Entity/Player/Player.cs:
```diff
@@ -19,7 +20,7 @@ public class Player : Entity
     {
         SetEnabled(true);
-        SetWeapon(weapon);
+        //SetWeapon(weapon);
```
Why comment this out? Did you forget to uncomment after debugging or? :)

Jazy5552 replied:
The SetWeapon, and all weapon logic, should now live in the ShootController. I simply commented it out to reduce any conflicts and allow for a refactor later.

AleksandrHovhannisyan replied:
Ahh, right right

AleksandrHovhannisyan commented on ProjectSettings/InputManager.asset:
```diff
@@ -37,6 +37,102 @@ InputManager:
+    negativeButton: joystick button 0
+    positiveButton: joystick button 3
+    altNegativeButton: joystick button 16
+    altPositiveButton: joystick button 19
```
What are buttons 16 and 19? They don't appear in the chart that Cameran linked a while ago.

Jazy5552 replied:
They are the same xbox controller buttons, but on MAC. http://wiki.unity3d.com/index.php/Xbox360Controller

AleksandrHovhannisyan replied:
Gotcha, so we're supporting both Mac and Windows basically? Or does this only support Mac atm?

AleksandrHovhannisyan commented on ProjectSettings/InputManager.asset:
```diff
@@ -37,6 +37,102 @@ InputManager:
+    descriptiveNegativeName: 
+    negativeButton: joystick button 2
+    positiveButton: joystick button 1
+    altNegativeButton: joystick button 18
```
Same here, Unity doesn't have any joystick button > 9.

Jazy5552 replied:
For MAC, http://wiki.unity3d.com/index.php/Xbox360Controller

Jazy5552 commented:
Yeha, I'm not a fan of the momentum at all. Just figured I would implement a simple checkbox so we could check it out. The whole, moving slower while player moves backwards, could easily be resolved by projecting the velocity vector alone the right vector of the rigidbody. I did fix it, just so you can see how to feels.

Jazy5552 committed:
Use more descriptive variable names

Jazy5552 committed:
Only include left and right momentum on the projectile

Jazy5552 committed:
Invert controller shooting axis

Jazy5552 committed:
Resolve merge conflicts with develop and feature/shoot-controls

Merge branch 'develop' of https://github.com/cap4053-cheeky-pixels/EmbodyGame into feature/shoot-controls

 Conflicts:
	Assets/Scripts/Entity/Entity.cs
	Assets/Scripts/Weapons/BasicWeapon.cs

Jazy5552 commented:
Merging due to previous review.
